src/cpp/rtps/messages/RTPSMessageGroup.cpp

```diff
@@ -264,9 +324,13 @@ RTPSMessageGroup::~RTPSMessageGroup() noexcept(false)
 
 void RTPSMessageGroup::reset_to_header()
 {
-    CDRMessage::initCDRMsg(full_msg_);
-    full_msg_->pos = RTPSMESSAGE_HEADER_SIZE;
-    full_msg_->length = RTPSMESSAGE_HEADER_SIZE;
+    CDRMessage::initCDRMsg(header_msg_);
+    header_msg_->pos = RTPSMESSAGE_HEADER_SIZE;
+    header_msg_->length = RTPSMESSAGE_HEADER_SIZE;
+
+    buffers_to_send_.clear();
+    copied_messages_.clear();
+    buffers_bytes_ = 0;
```

Comment:
Should we reset other attributes, like `pending_buffer_`, and `pending_padding_`?

Reply:
Not really, in `insert_submessage` we check if the current CDR message has enough space to accommodate the new message that we are trying to add. If there is not enough space, we will call `flush` to send the current message and insert the new one, thus, calling `reset_to_header`. If we reset `pending_buffer_` we will lost this payload. `pending_buffer_` and `pending_padding_` are reseted every time they are used to add a new buffer in `append_submessage` and before adding a new one in `addSubmessageData/DataFrag`. The latter can be skipped after using the rule of 0 in NetworkBuffer.